compiler/aarch64/codegen/UnaryEvaluator.cpp

```diff
@@ -81,17 +81,35 @@ TR::Register *OMR::ARM64::TreeEvaluator::lnegEvaluator(TR::Node *node, TR::CodeG
 	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
 	}
 
+static TR::Register *commonIntegerAbsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *reg = cg->gprClobberEvaluate(firstChild);
+   TR::Register *tempReg = cg->allocateRegister();
+
+   bool is64bit = node->getDataType().isInt64();
+   TR::InstOpCode::Mnemonic eorOp = is64bit ? TR::InstOpCode::eorx : TR::InstOpCode::eorw;
```

Comment:
While this doesn't need to be fixed here, it might be worth considering some helpers somewhere in the opcode implementation which will return the right opcode for an operation based on a width. We have some of those in the x86 where you can ask for a load or other operator for a given width which centralizes the opcode selection and reduces the ternary operators in the codegen code. It also provides a central place to update opcode choices later if needed. Again, more of a suggestion as the aarch64 implementation moves forward rather than saying this code itself should not be accepted.

Reply:
I agree with this suggestion.  It's been handy on x86 for sure.  @knn-k , please open an issue for this so it can get on the backlog.

Reply:
Opened Issue #3269.